src/diffusers/models/unet_2d_blocks.py

```diff
@@ -1001,6 +1002,10 @@ def custom_forward(*inputs):
 
             output_states = output_states + (hidden_states,)
 
+        if additional_residuals is not None:
+            hidden_states += additional_residuals
+            output_states = output_states[:-1] + (output_states[-1] + additional_residuals,)
```

Comment:
`additional_residuals` is added to `output_states[-1]` twice here no? is this intended?

Reply:
this is not intented, this implementation is incorrect. the second line is required only in case the first one looks like this: 
`hidden_states = hidden_states + additional_residuals`, previously unet had some problems with graph breaks, which slowdown torch.compile, so I suggested the change, but the author implemented it only partially 😆

Reply:
ah yes super sorry, I moved the application of the residual to inside the loop before output_states is appended to to make it clearer